test/models/media_resource_test.rb

```diff
@@ -128,16 +128,40 @@
 
   it "marks completed resources for replacement" do
     mr = build_stubbed(:media_resource, status: "started")
+    refute mr.marked_for_destruction?
     refute mr.marked_for_replacement?
 
     mr.mark_for_replacement
+    assert mr.marked_for_destruction?
     refute mr.marked_for_replacement?
 
     mr.status = "complete"
     mr.mark_for_replacement
+    assert mr.marked_for_destruction?
     assert mr.marked_for_replacement?
+  end
 
-    mr.replaced_at = nil
-    refute mr.marked_for_replacement?
+  it "sets both deleted_at and replaced_at on save" do
```

Comment:
Wrote some actual tests that create/delete stuff in the DB.  The stubbed tests weren't really covering this bug.